src/xcode/ENA/ENA/Source/HTTPClient/Service/ServiceError.swift

```diff
@@ -8,6 +8,16 @@ import ENASecurity
 The errors that can occur while using the service and calling http methods.
 */
 enum ServiceError<RE>: Error, Equatable where RE: Error {
+	// we need this property for nested errors
+	var errorDescription: String {
```

Comment:
Does it also work for the other nested errors?

Reply:
Currently, we need to exact error type for this specific case receivedResourceError, so it is currently confined to this case only. The implementation will remain untouched for other cases.

Reply:
Just to make sure if we all are on same page, it will cover all the messages for Ticket Validation.

Reply:
What about trustEvaluationError?

Reply:
This is beyond the scope of this bug ticket.

Reply:
The trust evaluation errors are usually mapped to custom errors with the right prefix, so it should work fine for the TicketValidation scope at least, I think

Reply:
Yes with this implementation, the short errors only work, if the error is mapped to receivedResourceError. It will not work with not mapped errors or other errors like invalidRequestError, resourceError.

Reply:
I have added the case for trust evaluation error.

Reply:
I think it makes sense to keep e.g. invalidRequestError and resourceError as it is so the outer error is visible so we can easily distinguish them (both have a ResourceError inside). There is no defined separate error code for them. From my perspective it should be okay to keep longer error description in these cases because they are not foreseen by the tech spec and we might need all the info we can get to debug.